.rustfmt.toml

```diff
@@ -1,67 +1,74 @@
-max_width = 90 # changed
-hard_tabs = false
-tab_spaces = 4
-newline_style = "Auto"
-use_small_heuristics = "Default"
-indent_style = "Block"
-wrap_comments = false
-format_code_in_doc_comments = false
-comment_width = 80
-normalize_comments = true # changed
-normalize_doc_attributes = false
-# license_template_path = "LICENSE_TEMPLATE" # changed
-format_strings = false
-format_macro_matchers = false
-format_macro_bodies = true
+# Most of the changed settings had defaults made for backcompat with earlier editions.
+# Brief descriptions are provided for nontrivial changes, otherwise a simple "changed" is
+# used.
+
+# `rustfmt` internal settings
+#
+# emit_mode = "Files"
+# make_backup = false
+
+binop_separator = "Front"
+blank_lines_lower_bound = 0
+blank_lines_upper_bound = 1
+brace_style = "SameLineWhere"
+color = "Auto"
+combine_control_expr = false # better all around in our opinions
+comment_width = 80 # changed
+condense_wildcard_suffixes = true # changed
+control_brace_style = "AlwaysSameLine"
+disable_all_formatting = false
+edition = "2018" # changed
 empty_item_single_line = true
-struct_lit_single_line = true
+enum_discrim_align_threshold = 0
+error_on_line_overflow = true # be more strict
+error_on_unformatted = true # be more strict
+fn_args_layout = "Tall"
+# unchanged, this does not work well with the long expressions of this crate
 fn_single_line = false
-where_single_line = false
+force_explicit_abi = true
+force_multiline_blocks = true # help certain closures
+format_code_in_doc_comments = true # changed
+format_macro_bodies = true # changed
+format_macro_matchers = true # changed
+format_strings = true # changed
+hard_tabs = false
+hide_parse_errors = false
+ignore = []
 imports_indent = "Block"
-imports_layout = "Vertical" # changed
+imports_layout = "Vertical" # makes editing quicker
+indent_style = "Block"
+inline_attribute_width = 0 # unchanged, for ease of editing
+# license_template_path = "LICENSE_TEMPLATE" # changed
+match_arm_blocks = true
+match_block_trailing_comma = false
+max_width = 90 # changed
+merge_derives = true
 merge_imports = true # changed
+newline_style = "Unix" # changed, prevents problems when a file is created on Windows
+normalize_comments = true # no /* */ comments should be in master
+normalize_doc_attributes = true # changed
+overflow_delimited_expr = false
+remove_nested_parens = true
+reorder_impl_items = true # changed
 reorder_imports = true
 reorder_modules = true
-reorder_impl_items = false
-type_punctuation_density = "Wide"
-space_before_colon = false
+report_fixme = "Never"
+report_todo = "Never"
```

Comment:
We should actually report them always. At least I would want that because I would not like to merge any PRs that actually have those in them. Instead convert TODO and FIXME into proper issues with proper description before merging a PR. This is a really nice to have warning.

Reply:
ok